app/assets/javascripts/search.js

```diff
@@ -0,0 +1,101 @@
+;(function (global) {
+  'use strict'
+
+  var $ = global.jQuery
+
+  var SearchResultsContainer = (function () {
+    var $searchResultContainer
+    var $results
+
+    var _setSearchDataAttributesOnResults = function () {
+      /*
+        Add the lowercased search terms to a 'data-search' attribute
+        for each possible search result
+       */
+      $results.each(function () {
+        var searchTerms = $(this).find('h2').text()
+        $(this).attr('data-search', searchTerms.toLowerCase())
+      })
+    }
+
+    var init = function () {
+      $searchResultContainer = $('#search-results')
+      $results = $searchResultContainer.find('.metric-group')
+
+      _setSearchDataAttributesOnResults()
+    }
+
+    var filter = function (query) {
+      /*
+        Hide search results unless the query substring matches the title
+       */
+      var matches = 0
```

Comment:
I think we're going need a "number of results" counter somewhere on the page, but at the moment this isn't used.